rabies/analysis_pkg/analysis_functions.py

```
@@ -242,192 +238,69 @@
+        setattr(self, 'CPCA_extra_filename', CPCA_extra_filename)
+        setattr(self, 'CPCA_report', report_folder)
 
         return runtime
 
     def _list_outputs(self):
-        return {'NPR_prior_timecourse_csv': getattr(self, 'NPR_prior_timecourse_csv'),
-                'NPR_extra_timecourse_csv': getattr(self, 'NPR_extra_timecourse_csv'),
-                'NPR_prior_filename': getattr(self, 'NPR_prior_filename'),
-                'NPR_extra_filename': getattr(self, 'NPR_extra_filename'),
-                'optimize_report': getattr(self, 'optimize_report'),
+        return {'CPCA_prior_timecourse_csv': getattr(self, 'CPCA_prior_timecourse_csv'),
```

Ruff (0.11.9)

Do not call `getattr` with a constant attribute value. It is not any safer than normal property access.

Replace `getattr` with attribute access

(B009)